plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -177,6 +185,55 @@ public Database getDatabase(String dbName)
         return client.getDatabase(dbName);
     }
 
+    @Override
+    public List<SchemaTableName> getAllTables()
+            throws TException
+    {
+        return alternativeCall(
+                exception -> !isUnknownMethodExceptionalResponse(exception),
+                chosenGetAllTablesAlternative,
+                () -> parseTableMetadata(client.getTableMeta("*", "*", ImmutableList.of())),
+                // fallback to fetching tables per schema if getTableMeta call is not available
+                () -> {
```

Comment:
this is a composite operation and each of the components should be retried individually.

you can change `io.trino.plugin.hive.metastore.thrift.ThriftMetastoreClient#getAllViews()` to return `Optional<List<>>` to indicate not supported